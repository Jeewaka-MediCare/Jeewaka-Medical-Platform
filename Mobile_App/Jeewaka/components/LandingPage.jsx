import React, { useEffect, useRef } from 'react';
import {
  View,
  Text,
  StyleSheet,
  Animated,
  Dimensions,
  Image,
  TouchableOpacity,
  ImageBackground
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useRouter } from 'expo-router';

const { width, height } = Dimensions.get('window');

export default function LandingPage() {
  const router = useRouter();
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const slideAnim = useRef(new Animated.Value(30)).current;
  const buttonScaleAnim1 = useRef(new Animated.Value(0.5)).current;
  const buttonScaleAnim2 = useRef(new Animated.Value(0.5)).current;

  useEffect(() => {
    console.log('Landing page animation starting...');
    
    // Reset values to ensure clean start
    fadeAnim.setValue(0);
    slideAnim.setValue(30);
    buttonScaleAnim1.setValue(0.5);
    buttonScaleAnim2.setValue(0.5);
    
    // Start main content animations
    const mainAnimations = Animated.parallel([
      Animated.timing(fadeAnim, {
        toValue: 1,
        duration: 800,
        useNativeDriver: true,
      }),
      Animated.spring(slideAnim, {
        toValue: 0,
        tension: 60,
        friction: 8,
        useNativeDriver: true,
      }),
    ]);

    mainAnimations.start(() => {
      console.log('Main animations completed');
      
      // Start button animations after main content
      Animated.sequence([
        Animated.spring(buttonScaleAnim1, {
          toValue: 1,
          tension: 80,
          friction: 6,
          useNativeDriver: true,
        }),
        Animated.spring(buttonScaleAnim2, {
          toValue: 1,
          tension: 80,
          friction: 6,
          useNativeDriver: true,
        }),
      ]).start(() => {
        console.log('All animations completed');
      });
    });
  }, []);

  const handleSignIn = () => {
    router.push('/login');
  };

  const handleCreateAccount = () => {
    router.push('/register');
  };

  return (
    <View style={styles.container}>
      {/* Background Image with Fade Effect */}
      <ImageBackground
        source={require('../assets/images/homeimage.png')} // Using existing image
        style={styles.backgroundImage}
        imageStyle={styles.backgroundImageStyle} // Add this line
        resizeMode="contain" // Changed from contain to cover for better positioning
      >
        {/* Fade Overlay for blurred effect */}
        <View style={styles.fadeOverlay} />
        
        {/* Content Container */}
        <View style={styles.content}>
          {/* Top Section - Text Content */}
          <View style={styles.topSection}>
            {/* Title */}
            <Animated.View
              style={[
                styles.titleContainer,
                {
                  opacity: fadeAnim,
                  transform: [{ translateY: slideAnim }]
                }
              ]}
            >
              <Text style={styles.subtitle}>Your Health, Our Priority</Text>
            </Animated.View>

            {/* Welcome Message */}
            <Animated.View
              style={[
                styles.welcomeContainer,
                {
                  opacity: fadeAnim,
                  transform: [{ translateY: slideAnim }]
                }
              ]}
            >
              <Text style={styles.welcomeTitle}>Welcome Back!</Text>
              <Text style={styles.welcomeText}>
                We're excited to see you again. Please sign in to continue your healthcare journey.
              </Text>
            </Animated.View>
          </View>

          {/* Bottom Section - Action Buttons */}
          <View style={styles.bottomSection}>
            <View style={styles.buttonsContainer}>
              <Animated.View
                style={[
                  styles.buttonWrapper,
                  {
                    opacity: fadeAnim,
                    transform: [{ scale: buttonScaleAnim1 }]
                  }
                ]}
              >
                <TouchableOpacity style={styles.primaryButton} onPress={handleSignIn}>
                  <Text style={styles.primaryButtonText}>Sign In</Text>
                </TouchableOpacity>
              </Animated.View>

              <Animated.View
                style={[
                  styles.buttonWrapper,
                  {
                    opacity: fadeAnim,
                    transform: [{ scale: buttonScaleAnim2 }]
                  }
                ]}
              >
                <TouchableOpacity style={styles.secondaryButton} onPress={handleCreateAccount}>
                  <Text style={styles.secondaryButtonText}>Create New Account</Text>
                </TouchableOpacity>
              </Animated.View>
            </View>
          </View>
        </View>
      </ImageBackground>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ffffffff', // Fallback background
  },
  backgroundImage: {
    flex: 1,
    width: '100%',
    height: '100%',
  },
  backgroundImageStyle: {
    transform: [
      { translateY: -90 } // Move UP by 30px (negative = up, positive = down)
    ],

  },
  fadeOverlay: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: 'rgba(255, 255, 255, 0.6)', // White overlay with 85% opacity for strong fade effect

  },
  content: {
    flex: 1,
    justifyContent: 'space-between', // This will push top and bottom sections apart
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingTop: 60, // Add top padding to push text higher
    paddingBottom: 100, // Add bottom padding to push buttons lower
    zIndex: 1, // Ensure content is above background
  },
  topSection: {
    top: 100,
    alignItems: 'center',
    flex: 0, // Don't let this section grow
  },
  bottomSection: {
    alignItems: 'center',
    flex: 0, // Don't let this section grow
    width: '100%',
  },
  titleContainer: {
    alignItems: 'center',
    marginBottom: 40,
  },
  title: {
    fontSize: 42,
    fontWeight: 'bold',
    color: '#027878ff', // Dark gray for better contrast on teal
    marginBottom: 8,
  },
  subtitle: {
    fontSize: 24, // Made larger since it's now the main title
    color: '#1b4641ff', // Dark gray for better visibility on teal
    fontStyle: 'italic',
    fontWeight: '600', // Added font weight
    textAlign: 'center',
  },
  imageContainer: {
    marginBottom: 30,
  },
  medicalImage: {
    width: width * 0.8,
    height: 200,
    borderRadius: 16,
    elevation: 5,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.15,
    shadowRadius: 12,
  },
  welcomeContainer: {
    alignItems: 'center',
    marginBottom: 30,
    paddingHorizontal: 20,
  },
  welcomeTitle: {
    fontSize: 28,
    fontWeight: '700',
    color: '#008080', // Dark gray for better contrast on teal
    marginBottom: 12,
  },
  welcomeText: {
    fontSize: 16,
    color: '#008080', // Medium gray for good readability on teal
    textAlign: 'center',
    lineHeight: 24,
  },
  buttonsContainer: {
    width: '100%', // Full width within bottomSection
    paddingHorizontal: 0, // Remove horizontal padding since bottomSection handles it
    alignItems: 'center',
  },
  buttonWrapper: {
    marginBottom: 12, // Reduced margin
    width: '80%', // Control button width here
  },
  primaryButton: {
    backgroundColor: '#008080', // Dark green complementing teal
    paddingVertical: 12, // Reduced padding
    paddingHorizontal: 24, // Added horizontal padding
    borderRadius: 8, // Smaller border radius
    alignItems: 'center',
    elevation: 2, // Reduced elevation
    shadowColor: '#008080',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 4,
  },
  primaryButtonText: {
    color: 'white',
    fontSize: 16, // Smaller font
    fontWeight: '600',
  },
  secondaryButton: {
    backgroundColor: 'transparent', // Changed to transparent
    paddingVertical: 12, // Reduced padding
    paddingHorizontal: 24, // Added horizontal padding
    borderRadius: 8, // Smaller border radius
    alignItems: 'center',
    borderWidth: 2,
    borderColor: '#047878ff', // Dark green border complementing teal
  },
  secondaryButtonText: {
    color: '#058686ff', // Dark green text complementing teal
    fontSize: 16, // Smaller font
    fontWeight: '600',
  },
});