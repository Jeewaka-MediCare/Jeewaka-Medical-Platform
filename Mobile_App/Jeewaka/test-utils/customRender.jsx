import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

// Custom render function that includes providers
export const renderWithProviders = (ui, options = {}) => {
  const Wrapper = ({ children }) => {
    return (
      <NavigationContainer>
        {children}
      </NavigationContainer>
    );
  };

  return render(ui, { wrapper: Wrapper, ...options });
};

// Re-export everything
export * from '@testing-library/react-native';

// Override render method
export { renderWithProviders as render };