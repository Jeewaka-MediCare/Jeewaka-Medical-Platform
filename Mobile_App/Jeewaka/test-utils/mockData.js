// Mock data for testing

export const mockDoctor = {
  _id: '507f1f77bcf86cd799439011',
  uid: 'doctor-uid-123',
  name: 'Dr. John Smith',
  email: 'john.smith@example.com',
  specialization: 'Cardiology',
  subSpecializations: ['Interventional Cardiology', 'Heart Surgery'],
  yearsOfExperience: 15,
  qualifications: ['MBBS', 'MD Cardiology', 'FRCPC'],
  hospitalAffiliation: 'General Hospital',
  consultationFee: 150,
  avgRating: 4.5,
  totalRatings: 120,
  totalReviews: 120,
  profilePicture: 'https://example.com/doctor.jpg',
  bio: 'Experienced cardiologist with 15 years of practice.',
  isAvailable: true,
  gender: 'Male',
  languages: ['English', 'Spanish'],
  phoneNumber: '+1234567890',
  address: '123 Medical St, City, State 12345',
  dateOfBirth: '1975-05-15',
  licenseNumber: 'MD12345',
  isVerified: true,
  createdAt: '2023-01-01T00:00:00.000Z',
  updatedAt: '2024-01-01T00:00:00.000Z',
};

export const mockPatient = {
  _id: '507f1f77bcf86cd799439012',
  uid: 'patient-uid-123',
  name: 'Jane Doe',
  email: 'jane.doe@example.com',
  phoneNumber: '+1234567891',
  dateOfBirth: '1990-03-20',
  gender: 'Female',
  address: '456 Patient Ave, City, State 12345',
  emergencyContact: {
    name: 'John Doe',
    relationship: 'Spouse',
    phoneNumber: '+1234567892',
  },
  medicalHistory: ['Diabetes', 'Hypertension'],
  allergies: ['Penicillin'],
  currentMedications: ['Metformin', 'Lisinopril'],
  insuranceInfo: {
    provider: 'Health Insurance Co',
    policyNumber: 'POL123456',
  },
  createdAt: '2023-01-01T00:00:00.000Z',
  updatedAt: '2024-01-01T00:00:00.000Z',
};

export const mockAppointment = {
  _id: '507f1f77bcf86cd799439013',
  patientId: mockPatient._id,
  doctorId: mockDoctor._id,
  date: '2024-02-15',
  startTime: '10:00 AM',
  endTime: '10:30 AM', 
  duration: 30,
  type: 'consultation',
  status: 'confirmed',
  consultationFee: 150,
  paymentStatus: 'paid',
  notes: 'Regular checkup',
  sessionId: 'session-123',
  doctorName: mockDoctor.name,
  doctorSpecialization: mockDoctor.specialization,
  patientName: mockPatient.name,
  isOngoing: false,
  createdAt: '2024-01-01T00:00:00.000Z',
  updatedAt: '2024-01-01T00:00:00.000Z',
};

export const mockPayment = {
  _id: '507f1f77bcf86cd799439014',
  appointmentId: mockAppointment._id,
  patientId: mockPatient._id,
  doctorId: mockDoctor._id,
  amount: 150,
  currency: 'LKR',
  paymentMethod: 'stripe',
  stripePaymentIntentId: 'pi_test_123',
  status: 'completed',
  doctorName: mockDoctor.name,
  doctorSpecialization: mockDoctor.specialization,
  date: '2024-01-01',
  appointmentDate: '2024-02-15',
  type: 'consultation',
  paidAt: '2024-01-01T12:00:00.000Z',
  createdAt: '2024-01-01T12:00:00.000Z',
};

export const mockReview = {
  _id: '507f1f77bcf86cd799439015',
  patientId: mockPatient._id,
  doctorId: mockDoctor._id,
  appointmentId: mockAppointment._id,
  rating: 5,
  comment: 'Excellent doctor, very professional and caring.',
  createdAt: '2024-01-02T00:00:00.000Z',
};

export const mockDoctorsList = [
  mockDoctor,
  {
    ...mockDoctor,
    _id: '507f1f77bcf86cd799439016',
    uid: 'doctor-uid-456',
    name: 'Dr. Sarah Johnson',
    email: 'sarah.johnson@example.com',
    specialization: 'Dermatology',
    subSpecializations: ['Cosmetic Dermatology', 'Pediatric Dermatology'],
    consultationFee: 120,
    avgRating: 4.8,
    totalRatings: 95,
    totalReviews: 95, // Add both fields for compatibility
  },
  {
    ...mockDoctor,
    _id: '507f1f77bcf86cd799439017',
    uid: 'doctor-uid-789',
    name: 'Dr. Michael Brown',
    email: 'michael.brown@example.com',
    specialization: 'Pediatrics',
    subSpecializations: ['Neonatology', 'Pediatric Cardiology'],
    consultationFee: 100,
    avgRating: 4.6,
    totalRatings: 87,
    totalReviews: 87, // Add both fields for compatibility
  },
];

export const mockApiResponse = {
  success: true,
  data: mockDoctorsList,
  message: 'Doctors fetched successfully',
};

export const mockApiError = {
  success: false,
  error: 'Network error',
  message: 'Failed to fetch data',
};

export const mockAuthUser = {
  uid: 'user-uid-123',
  email: 'test@example.com',
  displayName: 'Test User',
  emailVerified: true,
  getIdToken: jest.fn(() => Promise.resolve('mock-token')),
  getIdTokenResult: jest.fn(() => Promise.resolve({
    claims: { role: 'patient' }
  })),
};

export const mockMedicalRecord = {
  _id: '507f1f77bcf86cd799439018',
  patientId: mockPatient._id,
  doctorId: mockDoctor._id,
  appointmentId: mockAppointment._id,
  title: 'Cardiology Consultation',
  diagnosis: 'Hypertension',
  treatment: 'Prescribed medication and lifestyle changes',
  medications: ['Lisinopril 10mg', 'Amlodipine 5mg'],
  notes: 'Patient shows good response to treatment',
  labResults: 'Blood pressure: 130/80 mmHg',
  attachments: ['test-result.pdf'],
  date: '2024-02-15',
  createdAt: '2024-02-15T10:30:00.000Z',
  updatedAt: '2024-02-15T10:30:00.000Z',
};

export const mockVideoCallSession = {
  sessionId: 'session-123',
  participantId: 'participant-456',
  meetingId: 'meeting-789',
  status: 'ready',
  isConnected: false,
  localParticipant: null,
  participants: [],
  error: null,
};

export const mockAppointmentsList = [
  mockAppointment,
  {
    ...mockAppointment,
    _id: '507f1f77bcf86cd799439019',
    date: '2024-03-01',
    startTime: '2:00 PM',
    endTime: '2:30 PM',
    status: 'pending',
    paymentStatus: 'pending',
    isOngoing: false,
  },
  {
    ...mockAppointment,
    _id: '507f1f77bcf86cd799439020',
    date: '2024-01-15',
    startTime: '11:00 AM',
    endTime: '11:30 AM',
    status: 'completed',
    paymentStatus: 'paid',
    isOngoing: false,
  },
];

export const mockNavigationProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    reset: jest.fn(),
    setOptions: jest.fn(),
  },
  route: {
    params: {},
  },
};