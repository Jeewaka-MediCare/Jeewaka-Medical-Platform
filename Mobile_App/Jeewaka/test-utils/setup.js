import React from 'react';
import '@testing-library/jest-native/extend-expect';
import 'react-native-gesture-handler/jestSetup';

// Mock Firebase Auth - used by components for authentication
jest.mock('firebase/auth', () => ({
  getAuth: jest.fn(() => ({
    currentUser: null,
    onAuthStateChanged: jest.fn(),
  })),
  getReactNativePersistence: jest.fn(() => 'mockPersistence'),
  initializeAuth: jest.fn(() => ({
    currentUser: null,
    onAuthStateChanged: jest.fn(),
  })),
  onAuthStateChanged: jest.fn(),
  signInWithEmailAndPassword: jest.fn(),
  createUserWithEmailAndPassword: jest.fn(),
  signOut: jest.fn(),
}));

// Mock Firebase App
jest.mock('firebase/app', () => ({
  initializeApp: jest.fn(() => ({})),
  getApps: jest.fn(() => []),
  getApp: jest.fn(() => ({})),
}));

// Mock react-native-gifted-charts - used by components for charts
jest.mock('react-native-gifted-charts', () => ({
  LineChart: jest.fn(({ children, ...props }) => null),
  BarChart: jest.fn(({ children, ...props }) => null),
  PieChart: jest.fn(({ children, ...props }) => null),
}));

// Mock Expo modules
jest.mock('expo-router', () => ({
  useRouter: () => ({
    push: jest.fn(),
    replace: jest.fn(),
    back: jest.fn(),
    canGoBack: jest.fn(() => true),
  }),
  useLocalSearchParams: () => ({}),
  router: {
    push: jest.fn(),
    replace: jest.fn(),
    back: jest.fn(),
  },
  Stack: {
    Screen: ({ children, ...props }) => children || null,
  },
  Tabs: {
    Screen: ({ children, ...props }) => children || null,
  },
}));

jest.mock('expo-font', () => ({
  useFonts: () => [true],
}));

jest.mock('expo-status-bar', () => ({
  StatusBar: 'StatusBar',
}));

jest.mock('expo-splash-screen', () => ({
  hideAsync: jest.fn(),
  preventAutoHideAsync: jest.fn(),
}));

jest.mock('expo-constants', () => ({
  default: {
    expoConfig: {
      extra: {},
    },
  },
}));

// Mock AsyncStorage
jest.mock('@react-native-async-storage/async-storage', () => ({
  getItem: jest.fn(() => Promise.resolve(null)),
  setItem: jest.fn(() => Promise.resolve()),
  removeItem: jest.fn(() => Promise.resolve()),
  clear: jest.fn(() => Promise.resolve()),
}));

// Mock Reanimated
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  Reanimated.default.call = () => {};
  return Reanimated;
});

// Mock Safe Area Context
jest.mock('react-native-safe-area-context', () => ({
  SafeAreaProvider: ({ children }) => children,
  SafeAreaView: ({ children }) => children,
  useSafeAreaInsets: () => ({ top: 0, right: 0, bottom: 0, left: 0 }),
}));

// Mock Vector Icons - this handles all icon families used by components
jest.mock('@expo/vector-icons', () => {
  const React = require('react');
  const { Text } = require('react-native');
  
  const createMockIcon = (iconFamily) => {
    const MockIcon = (props) => 
      React.createElement(Text, { testID: `${iconFamily}-${props.name || 'icon'}` }, props.name || 'Icon');
    MockIcon.displayName = iconFamily;
    return MockIcon;
  };

  return {
    Ionicons: createMockIcon('Ionicons'),
    MaterialCommunityIcons: createMockIcon('MaterialCommunityIcons'),
    MaterialIcons: createMockIcon('MaterialIcons'),
    Feather: createMockIcon('Feather'),
    FontAwesome: createMockIcon('FontAwesome'),
    FontAwesome5: createMockIcon('FontAwesome5'),
    AntDesign: createMockIcon('AntDesign'),
    Entypo: createMockIcon('Entypo'),
    EvilIcons: createMockIcon('EvilIcons'),
    Foundation: createMockIcon('Foundation'),
    Octicons: createMockIcon('Octicons'),
    SimpleLineIcons: createMockIcon('SimpleLineIcons'),
    Zocial: createMockIcon('Zocial'),
  };
});

// Silence the warning about act() not being wrapped
const originalConsoleError = console.error;
beforeAll(() => {
  console.error = (...args) => {
    if (
      typeof args[0] === 'string' &&
      args[0].includes('Warning: ReactDOM.render is no longer supported')
    ) {
      return;
    }
    originalConsoleError.call(console, ...args);
  };
});

afterAll(() => {
  console.error = originalConsoleError;
});

// Global test timeout
jest.setTimeout(10000);