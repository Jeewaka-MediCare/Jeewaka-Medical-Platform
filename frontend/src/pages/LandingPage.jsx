import { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { Button } from "../components/ui/button";
import Hero from "../landing/Hero";
import LandingHeader from "../landing/LandingHeader";
import Features from "../landing/Features";
import HowItWorks from "../landing/HowItWorks";
import Statistics from "../landing/Statistics";
import DoctorSignup from "../landing/DoctorSignup";
import Footer from "../landing/Footer";

export default function LandingPage() {
  useEffect(() => {
    document.title = 'Jeewaka - Find a doctor';
  }, []);
  const navigate = useNavigate();

  return (
    <div className="min-h-screen bg-background">
      <LandingHeader />
      <Hero />
      <Features />
      <HowItWorks />
      <Statistics />
      <DoctorSignup />
      <Footer />
    </div>
  );
}