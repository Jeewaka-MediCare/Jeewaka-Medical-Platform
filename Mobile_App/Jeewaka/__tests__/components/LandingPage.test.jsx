import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { renderWithProviders } from '../../test-utils/customRender';
import LandingPage from '../../components/LandingPage';

// Mock expo-router
jest.mock('expo-router', () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

const mockPush = jest.fn();

describe('LandingPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { getByText } = renderWithProviders(<LandingPage />);
    
    expect(getByText('Your Health, Our Priority')).toBeTruthy();
    expect(getByText('Welcome Back!')).toBeTruthy();
    expect(getByText('Sign In')).toBeTruthy();
    expect(getByText('Create New Account')).toBeTruthy();
  });

  it('displays welcome message', () => {
    const { getByText } = renderWithProviders(<LandingPage />);
    
    expect(getByText("We're excited to see you again. Please sign in to continue your healthcare journey.")).toBeTruthy();
  });

  it('navigates to login when Sign In is pressed', () => {
    const { getByText } = renderWithProviders(<LandingPage />);
    
    const signInButton = getByText('Sign In');
    fireEvent.press(signInButton);
    
    expect(mockPush).toHaveBeenCalledWith('/login');
  });

  it('navigates to register when Create New Account is pressed', () => {
    const { getByText } = renderWithProviders(<LandingPage />);
    
    const registerButton = getByText('Create New Account');
    fireEvent.press(registerButton);
    
    expect(mockPush).toHaveBeenCalledWith('/register');
  });

  it('renders medical image', () => {
    const component = renderWithProviders(<LandingPage />);
    // Test that image component is rendered (React Native Testing Library limitation)
    // We test the presence of the component instead
    expect(component).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = renderWithProviders(<LandingPage />);
    expect(tree).toMatchSnapshot();
  });
});