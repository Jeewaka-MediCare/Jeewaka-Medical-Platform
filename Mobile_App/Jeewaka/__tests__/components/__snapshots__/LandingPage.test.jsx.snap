// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffffff",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "testUri": "../../../assets/images/homeimage.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": "100%",
          },
          {
            "transform": [
              {
                "translateY": -90,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        {
          "backgroundColor": "rgba(255, 255, 255, 0.6)",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-between",
          "paddingBottom": 100,
          "paddingHorizontal": 20,
          "paddingTop": 60,
          "zIndex": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 0,
            "top": 100,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "marginBottom": 40,
              "opacity": 0,
              "transform": [
                {
                  "translateY": 30,
                },
              ],
            }
          }
        >
          <Text
            style={
              {
                "color": "#1b4641ff",
                "fontSize": 24,
                "fontStyle": "italic",
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            Your Health, Our Priority
          </Text>
        </View>
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "marginBottom": 30,
              "opacity": 0,
              "paddingHorizontal": 20,
              "transform": [
                {
                  "translateY": 30,
                },
              ],
            }
          }
        >
          <Text
            style={
              {
                "color": "#008080",
                "fontSize": 28,
                "fontWeight": "700",
                "marginBottom": 12,
              }
            }
          >
            Welcome Back!
          </Text>
          <Text
            style={
              {
                "color": "#008080",
                "fontSize": 16,
                "lineHeight": 24,
                "textAlign": "center",
              }
            }
          >
            We're excited to see you again. Please sign in to continue your healthcare journey.
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "paddingHorizontal": 0,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "marginBottom": 12,
                "opacity": 0,
                "transform": [
                  {
                    "scale": 0.5,
                  },
                ],
                "width": "80%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#008080",
                  "borderRadius": 8,
                  "elevation": 2,
                  "opacity": 1,
                  "paddingHorizontal": 24,
                  "paddingVertical": 12,
                  "shadowColor": "#008080",
                  "shadowOffset": {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 4,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "600",
                  }
                }
              >
                Sign In
              </Text>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "marginBottom": 12,
                "opacity": 0,
                "transform": [
                  {
                    "scale": 0.5,
                  },
                ],
                "width": "80%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#047878ff",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "opacity": 1,
                  "paddingHorizontal": 24,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#058686ff",
                    "fontSize": 16,
                    "fontWeight": "600",
                  }
                }
              >
                Create New Account
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
